File: solutions/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum_=nums[0]
        max_sum=nums[0]
        for i in range(1,len(nums)):
            if sum_<0:
                sum_ = max(sum_, nums[i])
            else:
                sum_=sum_+nums[i]
            max_sum=max(sum_,max_sum)
        return max_sum
            
        

File: solutions/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum_ = Integer.MIN_VALUE;
        int sum_max=Integer.MIN_VALUE;
        for (int i: nums) {
            if (sum_<0) {
                sum_=Math.max(sum_,i);
            }
            else {
                sum_=sum_+i;
            }
            sum_max=Math.max(sum_,sum_max);
        }
     return sum_max;   
    }
}
